var/spack/repos/builtin/packages/py-petastorm/package.py

```diff
@@ -12,11 +12,13 @@ class PyPetastorm(PythonPackage):
     Tensorflow, Pytorch, and other Python-based ML training frameworks."""
 
     homepage = "https://github.com/uber/petastorm"
-    pypi = "petastorm/petastorm-0.8.2.tar.gz"
+    url = "https://github.com/uber/petastorm/archive/refs/tags/v0.11.4.tar.gz"
```

Comment:
PyPI no longer hosts source code, only wheels. Use GitHub for all versions for simplicity.